### PR TITLE
Fleshmind 

### PR DESCRIPTION
**Pull request by Gandalf2k15**
# Fleshmind

Welcome to the many, you are quite welcome here.

Fleshmind is a new random event that spawns a rogue flesh-hybrid AI that wants to do one thing: Spread, assimilate, further the agenda of the flesh, make you, one with the flesh.

It consists of a copious amount of new enemies with very unique abilities as well as challenging combat situations.

The crew directive of the random event is to stop the rogue AI by destroying all processor cores. After all processor cores are destroyed, the corruption will lose all sense of coordination, and will degrade. Mobs will start attacking eachother etc.

The AI has multiple levels and will level up as it grows bigger. With each level up, the AI becomes harder to defeat, new structures appear as well as new mobs.

The corruption can spread naturally or through the wirenet, so watch out.

## The AI
The AI, or controller, is what controls the hivemind.

## Structures
The AI has multiple structures to use to it's advantage.

## Mobs
The AI has multiple mobs to use to it's advantage.

## Wireweed
The lifeblood of the AI, this nasty stuff connects the components together.

Currently in ALPHA.

Credits for the sprites and sounds: https://github.com/discordia-space/CEV-Eris/pull/2894 (Macleod962)
Credits to Azarak for the controller code, which has been pretty heavily changed.

Because it's Eris I basically had to rewrite everything.

Still a work in progress, just checking CI, I cba to run it locally.

https://user-images.githubusercontent.com/9026500/171439180-967d8981-2d7a-4620-87a7-54cbcd065e80.mp4

## Changelog

:cl:
add: A new random event, Fleshmind.
/:cl:

### CODE REVIEW

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (Zonespace27) [inline _maps/_basemap.dm]: oops

**Comment by Gandalf2k15** [comment]
Wow. Passes CI. Nice.

**Comment by MassacreDemigirl** [comment]
Great work

**Comment by Nerev4r** [comment]
So, balance thoughts from the first round.
I think the ability for the fleshmind to transmit through pipes is probably the strongest part of it. Either it shouldn't rly be able to, or it should only be able to go to the nearest vent instead of just any of them.
Second, we could probably use a core weapon against wireweeds. Probably wirecutters. They take awhile to break.
Third, it could probably use an announcement? At least after five minutes after it spawns or something. Give it some grace period like blob.

Otherwise, some Nice Things.
It'd be cool for the wireweeds themselves to be able to convert people. Having to wait around for a mechiver is kinda lame, and people still get to control their bodies, so it shouldn't be too bad.
Whisperers could stand to be more common.
It'd be cool if you could willingly submit yourself for conversion, as I did write all those voicelines with the intent of even some people doing that.

It'd also be cool if it could turn carbons. Like, if you just had a funny overlay like Xyel mentioned but still had hands and shit.

image
As to bugs, these things are fucky. It doesn't attack windoors.

**Comment by YakumoChen** [comment]
hoping that the obvious system shock 2 voicelines are just a placeholder, I have reservations about using clearly copyrighted (and not open source) material on the server.

**Comment by Nerev4r** [comment]
Update: It's not spawning floaters.
Update: The death emotes from machines and such aren't appearing in the chatlog. Might want to give the death emotes/phrases to the mobs too?
Update (holy shit): 'Being "converted" pulls you from ghost roles or respawns. I got in as a SWAT officer but was yanked into my old body in the middle of it to be a robo-phaser!'

**Comment by YakumoChen** [comment]
I saw a floater when I was playing death commando, it did spawn, and exploded on me as intended.

**Comment by Nerev4r** [comment]
Scratch that, then. It doesn't spawn Whisperers, I don't believe.

**Comment by Gandalf2k15** [comment]
This is now in alpha.

**Comment by Tastyfish** [comment]
Can there be an antag icon like with families, cult etc for fellow infected? I had to squint or use examine to tell if some people were part of the fleshmind. Other people had the same problem.

**Comment by ghost** [comment]
was wondering what i kept thinking of when i saw this, its reminding me of Corruption from HZD <details><summary>Big image</summary>
image

If only 32x32 sprites could look this good :(</details>

So, if you wanted to try some light recoloring, that might be fun to aim for. Reddish/purplish corruption vibes.

---

From what I see in the PR itself, this just looks like a rethemed mold - not that thats a bad thing, though. Personally all for leaning into that, having it pseudo-containable and with key points to aim for.
Again just looking at the PR body, the messages seem... spam-my, they could probably be toned down or only activate on triggers (i.e. taking damage has a chance of certain messages, engaging a new target gives a message, etc), and the broken glass makes me feel sad because there's no reason for the fleshmind to have broken such pretty glass.
I can give better feedback after attending a test, but otherwise people seem to be enjoying it 👍 Just dont expect me to review the code LMAO

**Comment by Nerev4r** [comment]
I think the ending is a little unfitting; that is, nuking the place. You'd think the 'ending' would be maxing out the spread rate or something, so that it can convert as many people as possible rather than just killing them.

**Review by YakumoChen — CHANGES_REQUESTED** [review]
Remove the copyrighted sounds. I've played System Shock 2 before, and most of the mob sounds are identical or barely altered. Since you didn't claim to make them, the burden is on you to prove they aren't copyrighted.

**Comment by Hoolny** [comment]
Love it, cant wait to see it get stolen it into other servers

**Comment by Gandalf2k15** [comment]
Removed atmos from possible spawn locations

**Comment by Zenitheevee** [comment]
*bruh*

File: code/datums/ai/monkey/monkey_controller.dm [hunk 2]
```diff
@@ -152,6 +152,12 @@ have ways of interacting with a specific mob and control it.
 
 ///Reactive events to being hit
 /datum/ai_controller/monkey/proc/retaliate(mob/living/L)
+	// SKYRAT EDIT ADDITION
```
Comment (Zonespace27) [inline code/datums/ai/monkey/monkey_controller.dm]: this still needed?

File: modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	var/rocket_pod_charge_up_time = 3 SECONDS
+	/// How many rockets in our barage
+	var/barage = 1
+	/// How much time between rocket shots
+	var/barage_interval = 2
+	/// How often we can play the rotate sound
+	var/rotate_sound_cooldown_time = 1 SECONDS
+	COOLDOWN_DECLARE(rotate_sound_cooldown)
+	/// A list of footstep sounds we make
+	var/list/footstep_sounds = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm]: static

File: modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+	var/rocket_pod_cooldown_time_lower = 10 SECONDS
+	COOLDOWN_DECLARE(rocket_pod_cooldown)
+	/// The projectile we fire when shooting our rocket pods.
+	var/rocket_projectile_type = /obj/projectile/bullet/a84mm/weak
+	/// The sound we play when firing our rocket pods.
+	var/rocket_projectile_sound = 'sound/weapons/gun/general/rocket_launch.ogg'
+	/// The time it takes for us to charge up our rocket pods
+	var/rocket_pod_charge_up_time = 3 SECONDS
+	/// How many rockets in our barage
+	var/barage = 1
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm]: two "r"s

File: modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+	/// How much time between rocket shots
+	var/barage_interval = 2
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm]: ditto

File: modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+		'modular_skyrat/modules/fleshmind/sound/tyrant/footstep_6.ogg',
+	)
+	/// We also have a small laser to fire at people ;)
+	var/laser_cooldown_time_upper = 4 SECONDS
+	var/laser_cooldown_time_lower = 2 SECONDS
+	COOLDOWN_DECLARE(laser_cooldown)
+	/// Our laser projectile type
+	var/laser_projectile_type = /obj/projectile/beam/emitter/hitscan
+	/// A list of sounds we can play when firing the laser
+	var/list/laser_projectile_sounds = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_boss.dm]: static

File: modular_skyrat/modules/fleshmind/code/fleshmind_controller.dm [hunk 7]
```diff
@@ -0,0 +1,559 @@
...
+	// If no wireweed, spawn one under our first core.
+	if(!LAZYLEN(controlled_wireweed))
+		spawn_wireweed(get_turf(cores[1]), wireweed_type) // We use the first core in the list to spread.
+
+	// If no active wireweed, make all active and let the process figure it out.
+	if(!LAZYLEN(active_wireweed))
+		active_wireweed = controlled_wireweed.Copy()
+
+	var/list/spread_turf_canidates = list()
+	for(var/obj/structure/fleshmind/wireweed/wireweed as anything in active_wireweed)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_controller.dm:178]: this whole section could use a touch of documentation

File: modular_skyrat/modules/fleshmind/code/fleshmind_controller.dm [hunk 8]
```diff
@@ -0,0 +1,559 @@
...
+		message_admins("Corruption AI [controller_fullname] has been destroyed.")
+		SSshuttle.clearHostileEnvironment(src)
+
+/datum/fleshmind_controller/proc/end_game()
+	addtimer(CALLBACK(src, .proc/fleshmind_end_second_check), 20 SECONDS)
+	for(var/obj/structure/fleshmind/structure/core/iterating_core as anything in cores)
+		iterating_core.end_game = TRUE
+		iterating_core.update_appearance()
+	// Here we summon an ERT to defend the shuttle.
+	make_ert(/datum/ert/asset_protection, 5, "HEAD TO AND BOARD THE EMERGENCY SHUTTLE, DO NOT USE THE FERRY, STOP THE CORRUPTION AT **ALL** COSTS!!!", "the last defense of centcom", "GAMMA", FALSE, TRUE, TRUE, FALSE, TRUE)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_controller.dm]: this doesn't feel more than delta worthy

File: modular_skyrat/modules/fleshmind/code/fleshmind_environment_objects.dm [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+/obj/structure/fleshmind/wireweed/Initialize(mapload, starting_alpha = 255, datum/fleshmind_controller/incoming_controller)
+	. = ..()
+	alpha = starting_alpha
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = .proc/on_entered,
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+	our_controller = incoming_controller
+
+/obj/structure/fleshmind/wireweed/examine(mob/user)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_environment_objects.dm]: not needed

File: modular_skyrat/modules/fleshmind/code/fleshmind_environment_objects.dm [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+	/// The chance we have to ensnare a mob
+	var/ensnare_chance = 15
+	/// The amount of damage we do when attacking something.
+	var/object_attack_damage = 150
+	/// Are we active?
+	var/active = FALSE
+	/// Are we a vent burrow?
+	var/vent_burrow = FALSE
+	/// ZOnes we passively replace
+	var/list/replacement_zones = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_environment_objects.dm]: static

File: modular_skyrat/modules/fleshmind/code/fleshmind_event.dm [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+	qdel(resintest)
+
+	shuffle(turfs)
+	var/turf/picked_turf = pick(turfs)
+	if(!picked_turf)
+		message_admins("Fleshmind failed to spawn.")
+		return
+	var/obj/structure/fleshmind/structure/core/new_core = new(picked_turf)
+	announce_to_ghosts(new_core)
+
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_event.dm:51]:
```suggestion
```

File: modular_skyrat/modules/fleshmind/code/fleshmind_event.dm [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+/datum/round_event/fleshmind/proc/make_core()
+
+	var/list/turfs = list() //list of all the empty floor turfs in the hallway areas
+
+	var/obj/structure/biohazard_blob/resin/resintest = new()
+
+	var/list/possible_spawn_areas = typecacheof(typesof(/area/station/maintenance, /area/station/security/prison, /area/station/construction))
+
+	for(var/area/iterating_area in world)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_event.dm]: use GLOB.sortedAreas (or change the code a bit to use GLOB.areas_by_type)

File: modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm [hunk 13]
```diff
@@ -0,0 +1,1496 @@
...
+	var/stun_cooldown_time = 2 SECONDS
+	COOLDOWN_DECLARE(stun_cooldown)
+
+/mob/living/simple_animal/hostile/fleshmind/stunner/AttackingTarget(atom/attacked_target)
+	if(ishuman(target) && COOLDOWN_FINISHED(src, stun_cooldown))
+		var/mob/living/carbon/human/attacked_human = target
+		attacked_human.Knockdown(30)
+		playsound(src, 'sound/weapons/egloves.ogg', 50, TRUE)
+		COOLDOWN_START(src, stun_cooldown, stun_cooldown_time)
+	. = ..()
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm]:
```suggestion
	return ..()
```

File: modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm [hunk 14]
```diff
@@ -0,0 +1,1496 @@
...
+		'modular_skyrat/modules/fleshmind/sound/himan/passive_02.ogg',
+		'modular_skyrat/modules/fleshmind/sound/himan/passive_03.ogg',
+		'modular_skyrat/modules/fleshmind/sound/himan/passive_04.ogg',
+	)
+	del_on_death = TRUE
+	loot = list(
+		/obj/effect/gibspawner/human,
+	)
+	/// Are we currently faking our death? ready to pounce?
+	var/faking_death = FALSE
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm:748]: unsure about this, will have to see how it goes in practice

File: modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm [hunk 15]
```diff
@@ -0,0 +1,1496 @@
...
+				if(can_jump_on(nearby_turf, target_turf))
+					new_place = nearby_turf
+		else
+			new_place = target_turf
+
+	if(!new_place)
+		//there we make some kind of, you know, that creepy zig-zag moving
+		//we just take angle, distort it a bit and turn into dir
+		var/angle = get_angle(loc, target_turf)
+		angle += rand(5, 25)*pick(-1, 1)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm]:
```suggestion
		angle += rand(5, 25) * pick(-1, 1)
```

File: modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm [hunk 16]
```diff
@@ -0,0 +1,1496 @@
...
+	var/internal_mob_damage_upper = 60
+	/// Ditto
+	var/internal_mob_damage_lower = 40
+	/// How long we keep our passenger before either releasing or converting them.
+	var/conversion_time = 20 SECONDS
+	/// The comsume ability cooldown
+	var/consume_ability_cooldown_time = 1 MINUTES
+	COOLDOWN_DECLARE(consume_ability_cooldown)
+	/// A list of lines we will send to torment the passenger.
+	var/list/torment_lines = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_mobs.dm]: static

File: modular_skyrat/modules/fleshmind/code/human_corruption_component.dm [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+	/// A list of limbs we replace
+	var/list/replacement_zones = list(
+		BODY_ZONE_CHEST = /obj/item/bodypart/chest/robot,
+		BODY_ZONE_L_ARM = /obj/item/bodypart/l_arm/robot,
+		BODY_ZONE_L_LEG = /obj/item/bodypart/l_leg/robot,
+		BODY_ZONE_R_ARM = /obj/item/bodypart/r_arm/robot,
+		BODY_ZONE_R_LEG = /obj/item/bodypart/r_leg/robot,
+	)
+	/// These traits will be given and removed from the host.
+	var/list/traits_to_give = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/human_corruption_component.dm]: static

File: modular_skyrat/modules/fleshmind/code/human_corruption_component.dm [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+	var/static/list/actions_to_give = list(
+		/datum/action/cooldown/fleshmind_create_structure,
+		/datum/action/cooldown/fleshmind_create_structure/basic,
+		/datum/action/cooldown/fleshmind_plant_weeds,
+		/datum/action/cooldown/fleshmind_flesh_call,
+		/datum/action/innate/fleshmind_flesh_chat,
+	)
+	var/list/granted_actions = list()
+	/// A list of limbs we replace
+	var/list/replacement_zones = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/human_corruption_component.dm]: static

File: modular_skyrat/modules/fleshmind/code/machine_corruption_component.dm [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+#define PAIN_RESPONSE_SOUNDS list('modular_skyrat/modules/fleshmind/sound/robot_talk_heavy1.ogg', \
+	'modular_skyrat/modules/fleshmind/sound/robot_talk_heavy2.ogg', \
+	'modular_skyrat/modules/fleshmind/sound/robot_talk_heavy3.ogg', \
+	'modular_skyrat/modules/fleshmind/sound/robot_talk_heavy4.ogg',)
+
+#define MACHINE_TO_SPAWNER_PATHS list(/obj/machinery/rnd/production/techfab, /obj/machinery/autolathe, /obj/machinery/mecha_part_fabricator)
+
+/datum/component/machine_corruption
+	/// A list of possible overlays that we can choose from when we are created.
+	var/list/possible_overlays = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/machine_corruption_component.dm]: static

File: modular_skyrat/modules/fleshmind/code/make_ert.dm [hunk 20]
```diff
@@ -0,0 +1,121 @@
+#define ERT_EXPERIENCED_LEADER_CHOOSE_TOP 3
+
+/**
+ * Make ERT
+ *
+ * A generalised ERT creation proc, useful for when you want to make an ERT, but not require an admin.
+ *
+ * PROC IS NOT ASYNC.
+ */
+/proc/make_ert(ert_type, teamsize = 5, mission_objective_override = "Assist the station.", poll_description = "an ERT team", code = "UNKNOWN", enforce_human = FALSE, open_armory_doors = FALSE, leader_experience = FALSE, random_names = TRUE, notify_players = TRUE, spawnpoint_override)
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/make_ert.dm]: maybe move to somewhere more general-purpose?

File: modular_skyrat/modules/fleshmind/code/make_ert.dm [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+		earmarked_leader = pick(candidates)
+
+	var/index = 0
+	var/number_of_agents = min(created_ert_datum.teamsize, LAZYLEN(candidates))
+	var/team_spawned = FALSE
+
+	while(number_of_agents && LAZYLEN(candidates))
+		var/spawnloc = spawnpoints[index + 1]
+		//loop through spawnpoints one at a time
+		index = (index + 1) % spawnpoints.len
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/make_ert.dm]:
```suggestion
		index = (index + 1) % length(spawnpoints)
```

File: modular_skyrat/modules/fleshmind/code/make_ert.dm [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+		var/mob/dead/observer/chosen_candidate = earmarked_leader || pick(candidates) // this way we make sure that our leader gets chosen
+		candidates -= chosen_candidate
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/make_ert.dm]: use LAZYREMOVE (same goes for other list ops of `candidates` and `candidate_living_exps`

File: modular_skyrat/modules/fleshmind/code/make_ert.dm [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+	var/leader_spawned = FALSE // just in case the earmarked leader disconnects or becomes unavailable, we can try giving leader to the last guy to get chosen
+
+	if(created_ert_datum.leader_experience)
+		var/list/candidate_living_exps = list()
+		for(var/mob/dead/observer/potential_leader as anything in candidates)
+			candidate_living_exps[potential_leader] = potential_leader.client?.get_exp_living(TRUE)
+
+		candidate_living_exps = sort_list(candidate_living_exps, cmp = /proc/cmp_numeric_dsc)
+		if(LAZYLEN(candidate_living_exps) > ERT_EXPERIENCED_LEADER_CHOOSE_TOP)
+			candidate_living_exps = candidate_living_exps.Cut(ERT_EXPERIENCED_LEADER_CHOOSE_TOP + 1) // pick from the top ERT_EXPERIENCED_LEADER_CHOOSE_TOP contenders in playtime
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/make_ert.dm]: like here for example

File: modular_skyrat/modules/fleshmind/code/make_ert.dm [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+		team_spawned++
+
+	if(team_spawned)
+		message_admins("[created_ert_datum.polldesc] has spawned with the mission: [created_ert_datum.mission]")
+		if(created_ert_datum.notify_players)
+			priority_announce("Central command has responded to your request for a CODE [uppertext(created_ert_datum.code)] Emergency Response Team and have confirmed one to be enroute.", "ERT Request", ANNOUNCER_ERTYES)
+	//Open the Armory doors
+	if(created_ert_datum.opendoors)
+		for(var/obj/machinery/door/poddoor/ert/door in GLOB.airlocks)
+			door.open()
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/make_ert.dm]: may as well make these async so they don't open one
at
a
time

File: modular_skyrat/modules/fleshmind/code/fleshmind_structures.dm [hunk 25]
```diff
@@ -0,0 +1,738 @@
...
+			has_attacked = TRUE
+	else if(istype(thing, /obj/vehicle/sealed/mecha))
+		var/obj/vehicle/sealed/mecha/mecha_thing = thing
+		mecha_thing.take_damage(attack_damage, attack_damage_type, MELEE, 0, get_dir(mecha_thing, src))
+		has_attacked = TRUE
+	if(has_attacked)
+		thing.visible_message(span_warning("\The [src] strikes [thing]!"), span_userdanger("\The [src] strikes you!"))
+		playsound(loc, 'sound/effects/attackblob.ogg', 100, TRUE)
+		do_attack_animation(thing, ATTACK_EFFECT_PUNCH)
+		return TRUE
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_structures.dm:367]:
```suggestion
		return TRUE
	return .
```
tg coding standard when . is non-standard

File: modular_skyrat/modules/fleshmind/code/fleshmind_structures.dm [hunk 26]
```diff
@@ -0,0 +1,738 @@
...
+	icon_state = "orb"
+	base_icon_state = "orb"
+	max_integrity = 100
+	required_controller_level = CONTROLLER_LEVEL_2
+	/// Upper timer limit for our ability
+	automatic_trigger_time_upper = 1.5 MINUTES
+	/// Lower time limit for our ability.
+	automatic_trigger_time_lower = 40 SECONDS
+	/// A list of quotes we choose from to send to the player.
+	var/list/join_quotes = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_structures.dm]: static

File: modular_skyrat/modules/fleshmind/code/fleshmind_structures.dm [hunk 27]
```diff
@@ -0,0 +1,738 @@
...
+	density = FALSE
+	max_integrity = 260
+	activation_range = DEFAULT_VIEW_RANGE
+	ability_cooldown_time = 20 SECONDS
+	/// The max amount of mobs we can have at any one time.
+	var/max_mobs = 2
+	/// The current amount of spawned mobs
+	var/spawned_mobs = 0
+	/// The allowed monster types
+	var/list/monster_types = list(
```
Comment (Zonespace27) [inline modular_skyrat/modules/fleshmind/code/fleshmind_structures.dm]: static

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
cobalt wanted a review

File: modular_skyrat/modules/fleshmind/code/fleshmind_controller.dm (hunk 8 above)
Comment (Gandalf2k15) [reply modular_skyrat/modules/fleshmind/code/fleshmind_controller.dm]: It's gamma because in it's lore it has the power to destroy civilisations.

**Comment by Acer202** [comment]
This isn't gone for good, is it? It was super cool.

**Comment by Zonespace27** [comment]
> This isn't gone for good, is it? It was super cool.

It's mostly done, it just needs people to finish coding it

**Comment by Gandalf2k15** [comment]
Will re-open when ready to merge

**Comment by Acer202** [comment]
Heyhey it's me again. Skyrat RED is now a thing, prime for this kind of antag probably. Maybe this could live? (Please it was so cool)

**Comment by Gandalf2k15** [comment]
she's not dead yet captain

File: code/modules/unit_tests/simple_animal_freeze.dm [hunk 28]
```diff
@@ -166,6 +166,18 @@
...
+		/mob/living/simple_animal/hostile/fleshmind/floater,
+		/mob/living/simple_animal/hostile/fleshmind/globber,
+		/mob/living/simple_animal/hostile/fleshmind/stunner,
+		/mob/living/simple_animal/hostile/fleshmind/hiborg,
+		/mob/living/simple_animal/hostile/fleshmind/himan,
+		/mob/living/simple_animal/hostile/fleshmind/treader,
+		/mob/living/simple_animal/hostile/fleshmind/phaser,
+		/mob/living/simple_animal/hostile/fleshmind/mechiver,
+		/mob/living/simple_animal/hostile/fleshmind/mauler_monkey,
+		/mob/living/simple_animal/hostile/fleshmind,
```
Comment (StrangeWeirdKitten) [inline code/modules/unit_tests/simple_animal_freeze.dm:167]: These should probally be refactored before merge.

**Comment by Gandalf2k15** [comment]
TODO: add more abilities for the converted humans; tentacle and armblade. 
also level up stats for people to see
Investigate qdel issue with cores
Investigate crashing

**Comment by Detective-Google** [comment]
the mobs for this don't seem to properly be able to aggro onto and deal with sentient simplemobs a-la miner drake form or bear form.

**Comment by Gandalf2k15** [comment]
- [x] regen no fix blood
- [x] change ending
- [x] ODST to shuttle for defense
- [ ] make mechiver convert dead non-carbons to mutated flesh blobs
- [x] make armblade lesser
- [x] ending makes floor wireweed
- [ ] mechiver runs away when full
- [ ] tyrant can be called to location

**Comment by StrangeWeirdKitten** [comment]
I got around to refactoring half the AI mob code into basic mobs on the bubber end of this PR. https://github.com/Bubberstation/Bubberstation/pull/933

**Comment by Gandalf2k15** [comment]
someday

**Pull request closed without merging**